apps/content-publishing-api/test/app-v2-asset-upload.e2e-spec.ts

```diff
@@ -54,36 +73,300 @@ describe('AppController E2E v2 asset upload verification', () => {
       await app.close();
     });
     await app.init();
+
+    redis = app.get<Redis>(getRedisToken(DEFAULT_REDIS_NAMESPACE));
+  });
+
+  describe('/v1/assets/upload', () => {
+    it('upload request with no files should fail', async () => {
+      await request(app.getHttpServer()).put('/v1/asset/upload').expect(422);
+    });
+
+    it('too many files in a single upload request should fail', async () => {
+      let req = request(app.getHttpServer()).put('/v1/asset/upload');
+      for (const i of Array(11).keys()) {
+        req = req.attach('files', randomFile30K, `file${i}.jpg`);
+      }
+      await req.expect(400);
+    });
+
+    it('MIME type not in whitelist should fail', async () => {
+      await request(app.getHttpServer()).put('/v1/asset/upload').attach('files', randomFile30K, 'file.bmp').expect(422);
+    });
+
+    it('valid request with legal assets should work!', async () => {
+      await request(app.getHttpServer())
+        .put(`/v1/asset/upload`)
+        .attach('files', randomFile30K, 'file1.jpg')
+        .expect(202);
+    });
   });
-  it.skip('valid request with uploaded assets should work!', async () => {
-    const file = Buffer.from('h'.repeat(30 * 1000 * 1000)); // 30MB
-    const response = await request(app.getHttpServer())
-      .post(`/v2/asset/upload`)
-      .attach('files', file, 'file1.jpg')
-      .expect(202);
-    await sleep(1000);
-    const validReplyWithUploadedAssets = {
-      ...validReplyNoUploadedAssets,
-      content: {
-        ...validContentNoUploadedAssets,
-        assets: [
-          {
-            name: 'image asset',
-            references: [
-              {
-                referenceId: response.body.assetIds[0],
-                height: 123,
-                width: 321,
-              },
-            ],
+
+  describe('V1 Content and Profile', () => {
+    const missingAssetId = '9876543210';
+    const badMimeTypeReferenceId = '0123456789';
+
+    beforeAll(async () => {
+      await redis.set(badMimeTypeReferenceId, JSON.stringify(assetMetaDataInvalidMimeType));
+    });
+
+    describe('announcements with missing assets', () => {
+      it.each([
```

Comment:
this construction is new to me... table tests where appropriate make reviewing & debugging so much easier.